workflow/sim_params.py

```diff
@@ -0,0 +1,99 @@
+import os
+from typing import Union
+
+from qcore import utils
+
+
+def _update_params(d: dict, *u: dict) -> dict:
+    """
+    Update keys in a dictionary.
+
+    Prevents removal of keys in a nested dict. Note that the same key will
+    still be overwritten. The last dict in *u would preserve all its keys.
+
+    Parameters
+    ----------
+    d : dict
+        Original dictionary.
+    *u : dict(s)
+        Dictionary or dictionaries to update d with.
+
+    Returns
+    -------
+    dict
+        The dictionary d updated with the keys of *u.
+
+    Examples
+    --------
+    >>> a = {'hf': {'hf_dt': 1, 'x': 2}}
+    >>> b = {'hf': {'hf_dt': 3, 'y': 4}}
+    >>> a.update(b)
+    {'hf': {'hf_dt': 3, 'y': 4}}
+    >>> _update_params(a, b)
+    {'hf': {'hf_dt': 3, 'x': 2, 'y': 4}}
+    """
+    for uu in u:
+        if uu:  # if uu is not empty
+            for k, v in uu.items():
+                if isinstance(v, dict):
+                    d[k] = _update_params(d.get(k, {}), v)
+                else:
+                    d[k] = v
+    return d
+
+
+def load_sim_params(
+    sim_yaml_path: Union[bool, str] = False,
+    load_fault: Union[bool, str] = True,
+    load_root: Union[bool, str] = True,
+    load_vm: Union[bool, str] = True,
+) -> dict:
+    """Load all necessary parameters for a simulation.
+
+    Parameters
+    ----------
+    sim_yaml_path : Union[bool, str]
+        Path to sim_params.yaml or a falsy value to not load it.
```

Comment:
I think this should be falsy --> False ?

Reply:
No that's a [well-defined term](https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false).